src/test/java/org/opensearch/knn/index/LuceneEngineIT.java

```diff
@@ -338,6 +338,26 @@ public void testQuery_filterWithNonLuceneEngine() throws Exception {
         );
     }
 
+    public void testIndexReopening() throws Exception {
+        createKnnIndexMappingWithLuceneEngine(DIMENSION, SpaceType.L2);
+
+        for (int j = 0; j < TEST_INDEX_VECTORS.length; j++) {
+            addKnnDoc(INDEX_NAME, Integer.toString(j + 1), FIELD_NAME, TEST_INDEX_VECTORS[j]);
+        }
+
+        final float[] searchVector = TEST_QUERY_VECTORS[0];
+        final int k = 1;
+
+        validateQueryResultsQty(searchVector, k);
```

Comment:
```suggestion
        final int k = Math.random(10);
        List<KNNResult> expectedKnnResults = parseSearchResponse(responseBody, FIELD_NAME);
        closeIndex(INDEX_NAME);
        openIndex(INDEX_NAME);
        ensureGreen(INDEX_NAME);
        List<KNNResult> knnResults = parseSearchResponse(responseBody, FIELD_NAME);
        arrayEquals(expectedKnnResults, knnResults);
```

Reply:
I see you point, but why 10? Also this not gonna work as we may have less than 10 vectors and in such case number of results may be less than 10. Probably better

Reply:
Just an example. Just choose appropriate number please.